From bunk@stusta.de  Thu Aug  2 14:06:26 2007
From: Adrian Bunk <bunk@stusta.de>
Date: Tue, 31 Jul 2007 00:28:22 +0200
Subject: USB: Storage: usbat_check_status(): fix check-after-use
To: mdharm-usb@one-eyed-alien.net, gregkh@suse.de
Cc: linux-usb-devel@lists.sourceforge.net, usb-storage@lists.one-eyed-alien.net, linux-kernel@vger.kernel.org
Message-ID: <20070730222822.GG3972@stusta.de>
Content-Disposition: inline


The Coverity checker spotted that we have already oops'ed if "us"
was NULL.

Since "us" can't be NULL in the only caller this patch removes the
NULL check.

Signed-off-by: Adrian Bunk <bunk@stusta.de>
Signed-off-by: Matthew Dharm <mdharm-usb@one-eyed-alien.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/storage/shuttle_usbat.c |    3 ---
 1 file changed, 3 deletions(-)

--- a/drivers/usb/storage/shuttle_usbat.c
+++ b/drivers/usb/storage/shuttle_usbat.c
@@ -190,9 +190,6 @@ static int usbat_check_status(struct us_
 	unsigned char *reply = us->iobuf;
 	int rc;
 
-	if (!us)
-		return USB_STOR_TRANSPORT_ERROR;
-
 	rc = usbat_get_status(us, reply);
 	if (rc != USB_STOR_XFER_GOOD)
 		return USB_STOR_TRANSPORT_FAILED;
